idaes/core/util/model_diagnostics.py

```diff
@@ -3887,6 +3930,51 @@ def b_rule(b, i):
     return ill_cond
 
 
+def compute_infeasibility_explanation(
+    model, solver, tee=False, tolerance=1e-6, stream=None
```

Comment:
I think we should have `solver=None` here, and a check for:

```
if solver is None:
    solver = get_solver()
````

Reply:
Done in [40670e4](https://github.com/IDAES/idaes-pse/pull/1409/commits/40670e4dc0d66a766d87d4f70fb7820ecbf30965)